packages/core/src/components/Form.tsx

```diff
@@ -508,49 +520,70 @@ export default class Form<
 
     const mustValidate = !noValidate && liveValidate;
     let state: Partial<FormState<T, S, F>> = { formData, schema };
-    let newFormData = formData;
 
     if (omitExtraData === true && liveOmit === true) {
       const retrievedSchema = schemaUtils.retrieveSchema(schema, formData);
       const pathSchema = schemaUtils.toPathSchema(retrievedSchema, '', formData);
 
       const fieldNames = this.getFieldNames(pathSchema, formData);
 
-      newFormData = this.getUsedFormData(formData, fieldNames);
```

Comment:
Why not keep this change since the state update below on 541 will be the old form data before omitting extra data?

Reply:
Do you mean that this is a mistake, and L541 should get `newFormData`? I think this may be the case.